vendor/github.com/fluent/fluent-logger-golang/fluent/fluent.go

```diff
@@ -508,6 +520,15 @@ func (f *Fluent) run(ctx context.Context) {
 				return
 			}
 
+			if f.AsyncReconnectInterval > 0 {
+				if time.Since(f.latestReconnectTime) > time.Duration(f.AsyncReconnectInterval)*time.Millisecond {
```

Comment:
Didn't look too closely if it could be a problem, but wondering if the usage of `f.latestReconnectTime` also needs a lock here

(This is in vendored code, so if it needs fixing, needs to be done in the upstream project)

Reply:
I don't think it needs one, at least in the current code. The only place that calls `connect` (which assigns `latestReconnectTime` is this `if` loop (if AsyncReconnectInterval is enabled, otherwise it's the `writeWithRetry` below) -- concurrent calls to `connect` are not made). If you want to take a second look, I'm all ears. I'm definitely a less experienced Gopher than you, so perhaps I am missing something!

Reply:
>  concurrent calls to connect are not made

Ah, right, yes, in that case it won't be needed. I saw a mention of the lock in one of the functions, and wondered if things _could_ be called concurrently.

>  I'm definitely a less experienced Gopher than you, so perhaps I am missing something!

I still make mistakes and miss things as anyone! 😂 . What I _did_ learn over the years is that it usually doesn't hurt to leave a comment (even if wrong). I've been bitten by cases where I wasn't sure and "assumed" other participants on the PR would've noticed if it was wrong (because they are more experienced than I, so for sure they would see a mistake??) and it turned out there _was_ an issue after all, but everyone missed it.